### PR TITLE
feat: terminate transient queries by id

### LINKED ISSUE #5868: Transient queries cannot be terminated by QueryID
Some temp queries are running in the ksqldb. I cannot use "TERMINATE" to stop them. Please help on this. Thanks!

KSQL cli is 0.9.0 and KSQLDB is 5.5.

```
Query ID                                                            | Query Type | Status    | Sink Name | Sink Kafka Topic | Query String                                                               
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 _confluent-ksql-default_transient_4914033483836191816_1595404775294 | PUSH       | RUNNING:1 |           |                  | select * from WORKFLOW_INPUT_DOMAIN_SALESFORCE_ORDERS_STREAM emit changes; 
 _confluent-ksql-default_transient_2968417800293416691_1595412094814 | PUSH       | RUNNING:1 |           |                  | select * from WORKFLOW_INPUT_DOMAIN_SALESFORCE_ORDERS_STREAM emit changes; 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

In the ksqldb cli window. I input "terminate _confluent-ksql-default_transient_4914033483836191816_1595404775294;", it showed me like below:

```
**### line 1:21: mismatched input '-' expecting ';'
Statement: terminate _confluent-ksql-default_transient_4914033483836191816_1595404775294;
Caused by: line 1:21: mismatched input '-' expecting ';'
Caused by: org.antlr.v4.runtime.InputMismatchException**
```

### PR DESCRIPTION
**Pull request by nateab**
### Description 
New PR for #7805 

Allow users to terminate transient queries by their query id, which fixes #5868. This patch adds a new custom executor to handle these statements. Uses a scatter gather approach to be able to terminate any transient query in the cluster. Has the same limitation as `show queries` in that a persistent query must already be running on all nodes in order to find remote hosts, as described in #5105.

### Testing done 
Added integration tests `TerminateTransientQueryFunctionalTest`, added unit tests `TerminateQueryExecutorTest`, and updated existing unit tests.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-reference/quick-reference.md [hunk 1]
```diff
@@ -603,7 +603,7 @@ SPOOL <file_name|OFF>
 ```
 
 ## TERMINATE
-End a persistent query. For more information, see [SPOOL](../../ksqldb-reference/terminate).
+End a query. For more information, see [TERMINATE](../../ksqldb-reference/terminate).
```
Comment (JimGalasyn) [inline docs/developer-guide/ksqldb-reference/quick-reference.md:606]: Heh, thanks for fixing this!

**Review by JimGalasyn — APPROVED** [review]
LGTM!

File: docs/developer-guide/ksqldb-reference/quick-reference.md (hunk 1 above)
Comment (nateab) [reply docs/developer-guide/ksqldb-reference/quick-reference.md:606]: Haha no problem!

**Review by AlanConfluent — APPROVED** [review]
Reviewed previously here: https://github.com/confluentinc/ksql/pull/7805

**Pull request merged**